pkg/daemon/daemon.go

```diff
@@ -1290,12 +1359,66 @@ func (dn *Daemon) Run(stopCh <-chan struct{}, exitCh <-chan error) error {
 	for {
 		select {
 		case <-stopCh:
+			if dn.deferKubeletRestart {
+				// we also want to do this here just in case we miss a reboot
+				dn.deferKubeletRestart = false
+				err := os.Remove("/var/lib/kubelet/kubeconfig")
+				if err != nil {
+					return fmt.Errorf("could not remove kubelet's kubeconfig file: %v", err)
+				}
```

Comment:
What's the motivation behind this? This isn't something we've ever done before, and if you do it, then the Kubelet falls back to the bootstrap kubeconfig, which is not expected.
Kubelet should be renewing itself using the kubeconfig issued from the cluster, not the bootstrap kubeconfig,

The TLS bootstrap flow does not expect to ever have the kubelet re-issue a CSR request from a bootstrap kubeconfig if the Node object already exists.